### ISSUE TITLE
Route grade search, additional quick filter to only return routes using that grade system

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#1852]
You can already filter on the style independent of the grade, I think what we really want is just a tick box in the grade search saying 'Only find routes in this system'

I think this is simple than trying to detect or infer trad vs sport.

**Comment by brendanheywood** [comment]
Or if we are confidant that most people would actually expect only to see results of routes in this grade system we could make that the default behaviour and make the tick box 'Match equivalent grades in other systems'

**Comment by scd** [comment]
I am pretty sure people would want matches for all trad systems but not bouldering systems when they search for a trad climb of a particular system.

Not easy to search for the same system as the search is currently based on the internal system. Auto detecting is far easier.

**Comment by brendanheywood** [comment]
I don't think we have any reliable way to determine 'trad' from a grade, only from the style. Did mean the grade 'types' (I'm not really sure what we call these internally) ie what is shown on the grade article as in 'Free' 'Bouldering' 'Aid' 'Ice' 'Via ferrata' 'Mixed'. So if you simply meant 'Free' instead of trad then that will work.

So what do we call this type? and is this something that is indexed properly and we can easily search on via the facets?

This whole concept also will break down where ever the same grade system is used for both bouldering and free climbing, which is only a tiny edge case.

Perhaps a completely different solution is just reworking the UX so that the grade and style filters are close together and conceptually linked better so it's more clear that if you only want free climbing stuff then you just tick that in the style filter.

**Comment by scd** [comment]
+1

**Comment by brendanheywood** [comment]
Merging a duplicate from https://github.com/theCrag/website/issues/2567

This is to better inform the process of setting the most appropriate grade context for a crag, region or country. It would also be good to be able to see a list of routes whose context doesn't match it's crags context for review.

Further on the down the track the need for this may be diminished once we implement floating grades with on the fly conversions.

* [ ] filter to routes who are in a particular system
* [ ] filter to routes who are in a particular context
* [ ] filters to show all routes who have a context which is different to their parent or ancestor

**Comment by brendanheywood** [comment]
+1 via https://github.com/theCrag/website/issues/3046
